### PR TITLE
Reorganize cryprocurrency tests [ECR-782]

### PR DESCRIPTION
**Pull request by DarkEld3r**

### CODE REVIEW

File: examples/cryptocurrency/tests/constants/mod.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Shared constants used both in api and transactions logic tests.
+
+/// Alice's wallets name.
+pub const ALICE_NAME: &str = "Alice";
+/// Bob's wallet name.
+pub const BOB_NAME: &str = "Bob";
```
Comment (ozkriff) [inline examples/cryptocurrency/tests/constants/mod.rs]: Why `constants/mod.rs` instead of just `constants.rs`?

Comment (DarkEld3r) [reply examples/cryptocurrency/tests/constants/mod.rs]: https://doc.rust-lang.org/book/second-edition/ch11-03-test-organization.html#submodules-in-integration-tests 🙃

Comment (ozkriff) [reply examples/cryptocurrency/tests/constants/mod.rs]: Interesting official hack, didn't know about that.

File: examples/cryptocurrency/src/lib.rs [hunk 2]
```diff
@@ -210,20 +251,29 @@ impl Transaction for TxTransfer {
     /// [`TxCreateWallet`]: struct.TxCreateWallet.html
     fn execute(&self, view: &mut Fork) -> ExecutionResult {
```
Comment (vitvakatu) [inline examples/cryptocurrency/src/lib.rs]: Shouldn't we also check for `sender == receiver`?

Comment (DarkEld3r) [reply examples/cryptocurrency/src/lib.rs]: I'm not sure: this seems logically wrong, but technically gives no problem, right? We probably don't want to add unnecessary details to the tutorial.

@slowli What do you think?

Comment (slowli) [reply examples/cryptocurrency/src/lib.rs]: If I understand correctly, transactions with `sender == receiver` don't pass the `verify()` test, so during transaction execution, it is guaranteed that these two aren't equal.

File: examples/cryptocurrency/src/lib.rs [hunk 3]
```diff
@@ -172,6 +174,43 @@ transactions! {
...
+    ///
+    /// Can be emitted by `TxCreateWallet`.
+    #[fail(display = "Wallet already exists")]
+    WalletAlreadyExists = 0,
+
+    /// Sender doesn't exist.
+    ///
+    /// Can be emitted by `TxTransfer`.
+    #[fail(display = "Sender doesn't exist")]
+    SenderNotExist = 1,
```
Comment (slowli) [inline examples/cryptocurrency/src/lib.rs]: Maybe, use something like `AbsentSender` / `NonexistentSender` / `SenderNotFound`? `SenderNotExist` looks ungrammatic to me.

File: examples/cryptocurrency/src/lib.rs [hunk 4]
```diff
@@ -210,20 +251,29 @@ impl Transaction for TxTransfer {
...
+
+        let receiver = match schema.wallet(self.to()) {
+            Some(val) => val,
+            None => Err(Error::ReceiverNotExist)?,
+        };
+
+        let amount = self.amount();
+        if sender.balance() >= amount {
+            let sender = sender.decrease(amount);
+            let receiver = receiver.increase(amount);
```
Comment (slowli) [inline examples/cryptocurrency/src/lib.rs]: If I understand correctly, this instruction may lead to a panic (`u64` overflow, to be exact). I'm not quite sure whether this should be an error, or panicking is OK in this case.

File: examples/cryptocurrency/tests/constants/mod.rs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 The Exonum Team
```
Comment (slowli) [inline examples/cryptocurrency/tests/constants/mod.rs]: Nitpick: 2018?

File: examples/cryptocurrency/src/lib.rs (hunk 3 above)
Comment (DarkEld3r) [reply examples/cryptocurrency/src/lib.rs]: Originally it was named `SenderDoesntExist", but the spell-checker wasn't happy.
Fixed.

File: examples/cryptocurrency/src/lib.rs (hunk 4 above)
Comment (DarkEld3r) [reply examples/cryptocurrency/src/lib.rs]: As for me, panic is OK here.

**Pull request merged**